svgnative/ports/cairo/CairoSVGRenderer.cpp

```diff
@@ -0,0 +1,437 @@
+#elif CAIRO_HAS_RECORDING_SURFACE
+    cairo_surface_t* sf = cairo_recording_surface_create ( CAIRO_CONTENT_COLOR_ALPHA, NULL );
+#endif
+    mPath = cairo_create (sf);
+}
+
+CairoSVGPath::~CairoSVGPath()
+{
+    cairo_surface_t* sf = cairo_get_target( mPath );
+
+    cairo_destroy ( mPath );
+    cairo_surface_finish ( sf );
+    cairo_surface_destroy ( sf );
+}
+
+void CairoSVGPath::Rect(float x, float y, float width, float height)
+{
+    cairo_rectangle ( mPath, x, y, width, height );
+    cairo_close_path ( mPath );
+}
+
+inline double deg2rad(double deg)
+{
+    return (deg * M_PI / 180.0);
+}
+
+void CairoSVGPath::RoundedRect(float x, float y, float width, float height, float cornerRadius)
+{
+    // Cairo does not provide single API to draw "rounded rect". See
+    // https://www.cairographics.org/samples/rounded_rectangle/
+
+    cairo_new_sub_path ( mPath );
+    cairo_arc ( mPath, x - cornerRadius + width, y + cornerRadius,          cornerRadius, deg2rad(-90), deg2rad(  0));
+    cairo_arc ( mPath, x - cornerRadius + width, y - cornerRadius + height, cornerRadius, deg2rad(  0), deg2rad( 90));
+    cairo_arc ( mPath, x + cornerRadius,         y - cornerRadius + height, cornerRadius, deg2rad( 90), deg2rad(180));
+    cairo_arc ( mPath, x + cornerRadius,         y + cornerRadius,          cornerRadius, deg2rad(180), deg2rad(270));
+    cairo_close_path ( mPath );
+}
+
+void CairoSVGPath::Ellipse(float cx, float cy, float rx, float ry) {
+    // Cairo does not provide single API to draw "ellipse". See
+    // https://cairographics.org/cookbook/ellipses/
+
+    cairo_matrix_t  save_matrix;
+    cairo_get_matrix (mPath, &save_matrix); 
+
+    cairo_translate (mPath, cx, cy);
+    cairo_scale (mPath, rx, ry);
+    cairo_new_sub_path ( mPath );
+    cairo_arc (mPath, 0, 0, 1, 0, 2 * M_PI);
+
+    cairo_set_matrix ((cairo_t*)mPath, &save_matrix); 
+}
+
+void CairoSVGPath::MoveTo(float x, float y)
+{
+    cairo_move_to (mPath, x, y);
+    mCurrentX = x;
+    mCurrentY = y;
+}
+
+void CairoSVGPath::LineTo(float x, float y)
+{
+    cairo_line_to (mPath, x, y);
+    mCurrentX = x;
+    mCurrentY = y;
+}
+
+void CairoSVGPath::CurveTo(float x1, float y1, float x2, float y2, float x3, float y3)
+{
+    cairo_curve_to (mPath, x1, y1, x2, y2, x3, y3);
+    mCurrentX = x3;
+    mCurrentY = y3;
+}
+
+void CairoSVGPath::CurveToV(float x2, float y2, float x3, float y3)
+{
+    cairo_curve_to (mPath, mCurrentX, mCurrentY, x2, y2, x3, y3);
+    mCurrentX = x3;
+    mCurrentY = y3;
+}
+
+void CairoSVGPath::ClosePath() {
+    cairo_close_path (mPath);
+}
+
+CairoSVGTransform::CairoSVGTransform(float a, float b, float c, float d, float tx, float ty) {
+    cairo_matrix_init(&mMatrix, a, b, c, d, tx, ty);
+}
+
+void CairoSVGTransform::Set(float a, float b, float c, float d, float tx, float ty) {
+    // see CGSVGRenderer.cpp,
+    // internal of CGSVGTransform::CGSVGTransform() and CGSVGTransform::Set() are same.
```

Comment:
Not sure if this comment is needed or even helpful. IMO you can remove it.

Reply:
Indeed, I would remove this comment.